### ISSUE TITLE
stack overflow from `SVector(1,1)' .+ [1, 1]`

### ISSUE THREAD

**Issue by goretkin** [JuliaArrays/StaticArrays.jl#382]
On StaticArrays 0.7.0

```julia
julia> SVector(1,1)' .+ [1, 1]
ERROR: StackOverflowError:
Stacktrace:
 [1] broadcast_indices(::Type{T} where T, ::RowVector{Int64,StaticArrays.SArray{Tuple{2},Int64,1,2}}) at ./broadcast.jl:53
 [2] broadcast_indices(::RowVector{Int64,StaticArrays.SArray{Tuple{2},Int64,1,2}}) at ./broadcast.jl:48
 [3] broadcast_indices(::Type{T} where T, ::RowVector{Int64,StaticArrays.SArray{Tuple{2},Int64,1,2}}) at ./broadcast.jl:53
 [4] broadcast_indices(::RowVector{Int64,StaticArrays.SArray{Tuple{2},Int64,1,2}}) at ./broadcast.jl:48
 [5] broadcast_indices(::Type{T} where T, ::RowVector{Int64,StaticArrays.SArray{Tuple{2},Int64,1,2}}) at ./broadcast.jl:53
....
```

on StaticArrays 0.6.6

```
julia> SVector(1,1)' .+ [1, 1]
2×2 Array{Int64,2}:
 2  2
 2  2
```

**Comment by andyferris** [comment]
Hmm... we'll have to dig into this one...

Thanks for the report.

**Comment by tkoolen** [comment]
This is because StaticArrays defines these `_containertype` overloads:

https://github.com/JuliaArrays/StaticArrays.jl/blob/f7bf751eaa602b5dd8fef014c9750fd4247b604f/src/broadcast.jl#L14-L15

As of 0.6.2, the [`broadcast_indices`](https://github.com/JuliaLang/julia/blob/d386e40c17d43b79fc89d3e579fc04547241787c/base/broadcast.jl#L48) function in Base looks like
```julia
broadcast_indices(A) = broadcast_indices(containertype(A), A)
broadcast_indices(::ScalarType, A) = ()
broadcast_indices(::Type{Tuple}, A) = (OneTo(length(A)),)
broadcast_indices(::Type{Array}, A::Ref) = ()
broadcast_indices(::Type{Array}, A) = indices(A)
@inline broadcast_indices(A, B...) = broadcast_shape((), broadcast_indices(A), map(broadcast_indices, B)...)
```

Since `containertype(A)` is not one of the types for which Base defines methods, the top method calls the bottom one again. It's kind of an unfortunate design in Base that has been replaced in 0.7.

I'll submit a PR to fix 0.6.
